src/components/manager.js

```diff
@@ -57,6 +57,44 @@ const StyledTransition = styled(ReactTransitionGroup)({
   transformStyle: 'flat'
 });
 
+function buildSlideReference(props) {
```

Comment:
Could I ask why you opted to move this method outside the component?

Reply:
@kale-stew [getDerviedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) does not have access to the component instance. Since `buildSlideReference` didn't have any component instances in the method itself, I opted to take it out of the class and use it as a regular method that the class can use.